Sprint 2/step 1

heyxesha opened the pull request:
Промежуточный этап сдачи работы.
1. Работа BurgerConstructor перенесена в контекст.
2. Добавлены пустые состояния.
3. Отправляется запрос на получение номера заказа по клику на "Отправить заказ"
4. Заиспользован useReducer.
5. Всякие косметические правки конструктора.

heyxesha committed:
1. Перенесла состояние выбранных ингредиентов в контекст. Выбранные ингредиенты - пока что на моках.
BurgerConstructor теперь берет ингредтиенты из контекста.
2. Также подготовила пустые заглушки, когда не выбраны булки/начинки.
3. Понадобавляла классов, чтобы скролл появлялся только после определенного количества начинок + чтобы его появление смещало все элементы конструктора и чтобы было красивое выравнивание по правому краю.

heyxesha committed:
Перенесла рассчет суммы в useReducer.

heyxesha committed:
Добавила отправку POST запроса при клике на "Оформить заказ".
Попап берет номер заказа из ответа.

heyxesha committed:
Убрала запятушку.

andryxxa93 commented on src/App.jsx:
```diff
@@ -50,7 +53,9 @@ function App() {
                                 ) : (
                                     <>
                                         <BurgerIngredients ingredients={ state.ingredients } />
```
**Необходимо исправить**
Данные ингредиентов необходимо теперь передавать через контекст, а не через пропсы.

andryxxa93 reviewed (changes requested):
Отличная работа! 
Что понравилось:
- Контекст внедрен и использован верно
- Использован хук `useReducer`
- В контекст не обернуты лишние элементы

Есть только одно замечание, которое необходимо исправить, чтобы работа была принята:
- Необходимо полученные ингредиенты так же передавать с помощью контекста.

**Можно лучше**
- Размер модального окна при открытии номера заказа немного не соответствует тому, что в макете.

Удачных правок!

heyxesha committed:
Перенесла получение ингредиентов в BurgerIngredients в контекст.
В теории могла бы туда же и вызов API запихнуть, но учитывая, что заглушкой об ошибке хочется перекрывать оба компонента BurgerIngredients + BurgerConstructor (не давая пользователю ложных надежд), то пусть останется пока на уровне App.

heyxesha committed:
Привела размеры модалки к соответствию с макетом.

andryxxa93 reviewed (approved):
Отлично! Ваша работа принята.
Желаю удачного продолжения обучения!
